### PR TITLE
feat(gatsby): upgrade gatsby to 3.7.0

### PR DESCRIPTION
**Pull request by xiongemi**
## Current Behavior

currently gatsby is 3.5.0

## Expected Behavior

upgrade gatsby to 3.7.0

## Related Issue(s)

Fixes #

### CODE REVIEW

File: packages/gatsby/migrations.json [hunk 1]
```diff
@@ -145,6 +145,87 @@
           "alwaysAddToPackageJson": false
         }
       }
+    },
+    "12.3.7": {
+      "version": "12.7.0-beta.0",
```
Comment (FrozenPandaz) [inline packages/gatsby/migrations.json:150]: You probably meant `12.3.7-beta.0` :sweat_smile:

File: packages/gatsby/migrations.json [hunk 2]
```diff
@@ -145,6 +145,87 @@
...
+        },
+        "babel-plugin-module-resolver": {
+          "version": "4.1.0",
+          "alwaysAddToPackageJson": false
+        },
+        "babel-preset-gatsby": {
+          "version": "1.7.0",
+          "alwaysAddToPackageJson": false
+        },
+        "testingLibraryReactVersion": {
```
Comment (FrozenPandaz) [inline packages/gatsby/migrations.json:200]: This isn't a package is it?

**Review by FrozenPandaz — CHANGES_REQUESTED** [review]

File: packages/gatsby/migrations.json (hunk 1 above)
Comment (xiongemi) [reply packages/gatsby/migrations.json:150]: yes.

File: packages/gatsby/migrations.json (hunk 2 above)
Comment (xiongemi) [reply packages/gatsby/migrations.json]: yes. copied it wrong. changed.

File: packages/gatsby/migrations.json (hunk 1 above)
Comment (xiongemi) [reply packages/gatsby/migrations.json]: changed.

**Comment by jaysoo** [comment]
I'm getting the following warnings and errors in npm 7. 

@xiongemi can you take a look?

```npm WARN ERESOLVE overriding peer dependency
npm WARN Found: graphql@15.5.0
npm WARN node_modules/graphql
npm WARN   graphql@"^15.4.0" from gatsby@3.7.0
npm WARN   node_modules/gatsby
npm WARN     gatsby@"3.7.0" from the root project
npm WARN     10 more (gatsby-plugin-image, gatsby-plugin-sharp, ...)
npm WARN   1 more (eslint-plugin-graphql)
npm WARN
npm WARN Could not resolve dependency:
npm WARN peer graphql@"^14.4.1" from express-graphql@0.9.0
npm WARN node_modules/express-graphql
npm WARN   express-graphql@"^0.9.0" from gatsby@3.7.0
npm WARN   node_modules/gatsby
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR!
npm ERR! Found: @babel/core@7.12.13
npm ERR! node_modules/@babel/core
npm ERR!   dev @babel/core@"7.12.13" from the root project
npm ERR!   peer @babel/core@"^7.0.0-0" from @babel/preset-env@7.12.13
npm ERR!   node_modules/@babel/preset-env
npm ERR!     dev @babel/preset-env@"7.12.13" from the root project
npm ERR!     @babel/preset-env@"7.12.13" from @nrwl/web@200.0.0
npm ERR!     node_modules/@nrwl/web
npm ERR!       dev @nrwl/web@"200.0.0" from the root project
npm ERR!       1 more (@nrwl/react)
npm ERR!     2 more (@svgr/webpack, @cypress/webpack-preprocessor)
npm ERR!   111 more (@babel/preset-react, @babel/preset-typescript, ...)
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer @babel/core@"^7.13.0" from @babel/plugin-bugfix-v8-spread-parameters-in-optional-chaining@7.14.5
npm ERR! node_modules/babel-preset-gatsby/node_modules/@babel/plugin-bugfix-v8-spread-parameters-in-optional-chaining
npm ERR!   @babel/plugin-bugfix-v8-spread-parameters-in-optional-chaining@"^7.14.5" from @babel/preset-env@7.14.5
npm ERR!   node_modules/babel-preset-gatsby/node_modules/@babel/preset-env
npm ERR!     @babel/preset-env@"^7.14.4" from babel-preset-gatsby@1.7.0
npm ERR!     node_modules/babel-preset-gatsby
npm ERR!       dev babel-preset-gatsby@"1.7.0" from the root project
npm ERR!       1 more (gatsby)
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR!
npm ERR! See /Users/jack/.npm/eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/jack/.npm/_logs/2021-06-15T15_16_56_586Z-debug.log
```

**Pull request closed without merging**